### PR TITLE
Add lint and codecov workflow

### PR DESCRIPTION
**Pull request by mikebauer**
## Description of change
Add the modified lint and codecov workflow for re-core.

## Guide to reproduce test results.
Run the action in GH.
https://app.codecov.io/gh/resim-ai/re-core/tree/mikebauer%2Fcodecov

## Checklist:
- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [ ] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: .github/workflows/lint-and-codecov.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  push:
+    branches:
+      - main
+
+jobs:
+  coverage:
+    name: Lint & Generate Coverage Report
+    runs-on: ubuntu-20.04
+    container:
+      image: public.ecr.aws/m1r4u2a2/core:latest
```
Comment (iainjwhiteside) [inline .github/workflows/lint-and-codecov.yml:13]: shall we switch to the new friendly name?

File: .github/workflows/lint-and-codecov.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          path: |
+            ~/.cache/bazel
+          key: ${{ runner.os }}-bazel-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE') }}
+          restore-keys: |
+            ${{ runner.os }}-bazel-
+
+      - name: Build `compile_commands.json` database
+        run: bazel run @hedron_compile_commands//:refresh_all
+
+      - name: Workaround for Custom Docker Image
```
Comment (iainjwhiteside) [inline .github/workflows/lint-and-codecov.yml:30]: nano-nit: it might be worth a few works of a comment/description to explain why/what this is?

**Review by iainjwhiteside — APPROVED** [review]

File: .github/workflows/lint-and-codecov.yml (hunk 1 above)
Comment (mikebauer) [reply .github/workflows/lint-and-codecov.yml:13]: Good call. I'll replace this.

File: .github/workflows/lint-and-codecov.yml (hunk 2 above)
Comment (mikebauer) [reply .github/workflows/lint-and-codecov.yml:30]: I can try to figure out why. I just copy/pasted this tbh. 😅

Comment (mikebauer) [reply .github/workflows/lint-and-codecov.yml:30]: Good call. I've added some.

**Pull request merged**